index.html

```diff
@@ -161,123 +144,134 @@ <h2>
-    <li>Return <strong>undefined</strong>.</li>
-  </ol>
-
-  <p>The <dfn data-dfn-for="Performance" data-lt='clearMeasures'>clearMeasures(measureName)</dfn> removes stored timestamp with the associated name. It MUST run these steps:</p>
-  <ol>
-    <li>If <var>measureName</var> is omitted, remove all <a>PerformanceMeasure</a> objects in the <a href='https://w3c.github.io/performance-timeline/#dfn-performance-entry-buffer'>performance entry buffer</a>.</li>
-    <li>Otherwise remove all <a>PerformanceMeasure</a> objects listed in the <a href='https://w3c.github.io/performance-timeline/#dfn-performance-entry-buffer'>performance entry buffer</a> whose <code>name</code> matches <var>measureName</var>.</li>
-    <li>Return <strong>undefined</strong>.</li>
-  </ol>
-
-  </section>
-
-  <section id="performancemark">
-    <h2> The <code>PerformanceMark</code> Interface</h2>
-
-    <p>
-      The <dfn>PerformanceMark</dfn> interface also
-    exposes marks created via the <a>Performance.mark</a>
-    method to the
-      <a href="https://w3c.github.io/performance-timeline/#dfn-performance-timeline">Performance Timeline</a> [[!PERFORMANCE-TIMELINE-2]].
-    </p>
-
-    <pre class="idl">
-  [Exposed=(Window,Worker)]
-  interface PerformanceMark : PerformanceEntry {
-  };
-    </pre>
-
-    <p>
-    The <a>PerformanceMark</a>
-		interface extends the following attributes of the <a href="https://w3c.github.io/performance-timeline/#performanceentry">PerformanceEntry</a>
-		interface:
-    </p>
-
-        <p>The <code>name</code> attribute will return the mark's name.</p>
-        <p>The <code>entryType</code> attribute will return the <a href='https://www.w3.org/TR/WebIDL/#idl-DOMString'><code>DOMString</code></a> <code>"mark"</code>.</p>
-        <p>The <code>startTime</code> attribute will return a <a href="https://w3c.github.io/hr-time/#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-		with the mark's time value [[!HR-TIME-2]].
-    	</p>
-        <p>The <code>duration</code> attribute will return a <a href="https://w3c.github.io/hr-time/#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-		of value 0.</p>
+        partial interface Performance {
+            void mark(DOMString markName);
+            void clearMarks(optional DOMString markName);
+            void measure(DOMString measureName, optional DOMString startMark, optional DOMString endMark);
+            void clearMeasures(optional DOMString measureName);
+        };
+      </pre>
+      <section>
+        <h2><dfn>mark()</dfn> method</h2>
+        <p>Stores a timestamp with the associated name (a "mark"). It MUST run these steps:</p>
+        <ol>
+          <li>Run the <a href="#ensure-name-is-valid">ensure name is valid</a> algorithm with <var>name</var> set to <var>markName</var>.</li>
+          <li>Create a new <a>PerformanceMark</a> object.</li>
+          <li>Set <code>name</code> to <var>markName</var>.</li>

```

Comment:
This and following steps implicitly refer to the new object. Let's be explicit?

Reply:
Done.